PyRDF/Backend/Dist.py

```diff
@@ -665,7 +659,9 @@ def reducer(mergeables_out, mergeables_in):
                 for filename in value:
                     snapshot_chain.Add(filename)
                 # Create a new rdf with the chain and return that to user
-                snapshot_rdf = PyRDF.RDataFrame(snapshot_chain)
+                factory = PyRDF.DataFrame.DataFrameFactory()
+                snapshot_rdf = factory.create(
+                    snapshot_chain, backend=str(generator.head_node.backend))
```

Comment:
Ok, I would get rid of the factory, because we only need the backend anyway (the backend is our factory). So we can assume that in every backend implementation there will be a free function called `MakeDataFrame`. Such function is actually the one we will inject in `RDF` as `MakeBackendDataFrame`, but it's also accessible via the backend. Sound good?

So here we would do `generator.head_node.backend.MakeDataFrame(snapshot_chain)`